src/bgw/scheduler.c

```diff
@@ -325,19 +330,110 @@ scheduled_bgw_job_start(ScheduledBgwJob *sjob, register_background_worker_callba
 	}
 }
 
-static List *
-initialize_scheduled_jobs_list(MemoryContext mctx)
+static void
+terminate_and_cleanup_job(ScheduledBgwJob *sjob)
 {
-	List	   *jobs = bgw_job_get_all(sizeof(ScheduledBgwJob), mctx);
-	ListCell   *lc;
+	if (sjob->handle != NULL)
+	{
+		TerminateBackgroundWorker(sjob->handle);
+		WaitForBackgroundWorkerShutdown(sjob->handle);
+		pfree(sjob->handle);
```

Comment:
add sjob->handle = NULL for safety?

Reply:
done, and moved that to `worker_state_cleanup`